### PR TITLE
Support include rework for C* connector

### PR DESCRIPTION
**Pull request by Setogit**
#### Related issues

- https://github.com/strongloop/loopback-connector-cassandra/issues/8

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: test/include.test.js [hunk 1]
```diff
@@ -887,6 +1270,7 @@ describe('include', function() {
                   },
                   include: 'parts',
                 }, function(err, result) {
+                  console.log('=================== H-4:', err);
```
Comment (raymondfeng) [inline test/include.test.js]: The `console.log` should be removed from tests.

File: test/include.test.js [hunk 2]
```diff
@@ -53,14 +55,15 @@ describe('include', function() {
 
         u.__cachedRelations.should.have.property('posts');
         u.__cachedRelations.posts.forEach(function(p) {
-          p.userId.should.eql(u.id);
+          p.userId.toString().should.eql(u.id.toString());
```
Comment (raymondfeng) [inline test/include.test.js]: Why do we need this change?

File: test/include.test.js [hunk 3]
```diff
@@ -40,7 +41,8 @@ describe('include', function() {
     });
   });
 
-  it('should fetch hasMany relation', function(done) {
+  bdd.itIf(connectorCapabilities.supportNonPrimaryKeyIN !== false,
```
Comment (raymondfeng) [inline test/include.test.js]: I think we can fix the inclusion implementation to allow `=` instead of `IN`.

**Comment by raymondfeng** [comment]
We can fix https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/include.js#L255 to avoid `inq` if the length of FKs is 1.

File: test/include.test.js (hunk 1 above)
Comment (Setogit) [inline test/include.test.js]: done.  and several more console.log...

File: test/include.test.js (hunk 2 above)
Comment (Setogit) [inline test/include.test.js]: obsolete.  deleted.

Comment (Setogit) [inline test/include.test.js]: That's a left-over from trying the two designs back and force and there are many other similar cases.  At one point, made all the adhocSort === false cases to use string-based comparison with `should.be.equalOneOf` .

Did the following exercise: Changed `equalOneOf` to `oneOf` (`eql` equivalent) and removed all `toString()` except `toString() is required.

File: test/include.test.js (hunk 3 above)
Comment (Setogit) [inline test/include.test.js]: like this?
```
      newFilter.where[fkName] = foreignKeys.length === 1 ? foreignKeys[0] : {
        inq: foreignKeys,
      };
```

Comment (Setogit) [inline test/include.test.js]: For the full fledged lusion implementation to allow `=` , an issue has been filed as:
https://waffle.io/strongloop-internal/scrum-asteroid/cards/59135f6e0fe1c5009245818f

**Comment by Setogit** [comment]
Rebased/squasshed.  With inqLimit=1 change on the C* connector side, `include` cases are mostly passing.

**Comment by Setogit** [comment]
The removal of redundant `userId`restriction by @raymondfeng looks good to me; that's the right fix in the test case, I agree.  The related loopbck-connector PR: https://github.com/strongloop/loopback-connector/pull/103 has a bad side effect and with the test case fix, the #103 PR is not needed.  Closed the #103 PR.

**Pull request merged**